scheduler/pkg/agent/repository/triton/triton.go

```diff
@@ -42,35 +42,91 @@ func NewTritonRepositoryHandler(logger log.FieldLogger) *TritonRepositoryHandler
+		if !info.IsDir() && filepath.Base(path) != TritonConfigFile {
+			err := copy2.Copy(path, filepath.Join(dst, filepath.Base(path)))
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+	return err
+}
+
+// Copy config file to dst creating if required and setting the correct model name
+func (t *TritonRepositoryHandler) UpdateModelRepository(modelName string, rclonePath string, isVersionFolder bool, modelRepoPath string) error {
 	configFilePathRepo := filepath.Join(modelRepoPath, TritonConfigFile)
-	configFilePathFromVersion := filepath.Join(filepath.Dir(versionPath), TritonConfigFile)
-	if _, err := os.Stat(configFilePathFromVersion); err != nil {
-		return fmt.Errorf("Can't find %s", configFilePathFromVersion)
+	var configFilePath string
+	if isVersionFolder {
+		t.logger.Infof("Copy files from versioned folder %s to %s", filepath.Dir(rclonePath), modelRepoPath)
+		// copy all non-config.pbtxt files from folder above version to repo folder
+		err := copyNonConfigFilesToModelRepo(filepath.Dir(rclonePath), modelRepoPath)
+		if err != nil {
+			return err
+		}
+		// look for config.pbtxt in folder above of current folder if this is a version folder
+		configFilePath = filepath.Join(filepath.Dir(rclonePath), TritonConfigFile)
+		if _, err := os.Stat(configFilePath); err != nil {
+			// Create basic config.pbtxt as we didn't find it
+			return t.createConfigFileWithName(modelName, configFilePathRepo)
+		}
+	} else {
+		t.logger.Infof("Copy files from non-versioned folder %s to %s", rclonePath, modelRepoPath)
+		// copy all non-config.pbtxt files from folder to repo folder
+		err := copyNonConfigFilesToModelRepo(rclonePath, modelRepoPath)
+		if err != nil {
+			return err
+		}
+		// look for config.pbtxt in same folder as model artifacts
+		configFilePath = filepath.Join(rclonePath, TritonConfigFile)
+		if _, err := os.Stat(configFilePath); err != nil {
+			// Create basic config.pbtxt as we didn't find it
+			return t.createConfigFileWithName(modelName, configFilePathRepo)
+		}
 	}
+
+	// If we are here we found a config.pbtxt
 	// Always copy config.pbtxt overwriting existing as we may have changes in configuration
-	err := copy2.Copy(configFilePathFromVersion, configFilePathRepo)
+	err := copy2.Copy(configFilePath, configFilePathRepo)
 	if err != nil {
 		return err
 	}
 	return t.updateModelNameInConfig(modelName, configFilePathRepo)
 }
 
-func (t *TritonRepositoryHandler) updateModelNameInConfig(modelName string, path string) error {
-	s, err := t.loadConfigFromFile(path)
+func saveConfigFile(path string, config *pb.ModelConfig) error {
+	data, err := prototext.Marshal(config)
 	if err != nil {
 		return err
 	}
-	s.Name = modelName
-	data, err := prototext.Marshal(s)
+	return os.WriteFile(path, data, fs.ModePerm)
+}
+
+func (t *TritonRepositoryHandler) createConfigFileWithName(modelName string, path string) error {
+	config := pb.ModelConfig{}
+	config.Name = modelName
+	return saveConfigFile(path, &config)
+}
```

Comment:
question: are we relying on triton detecting the correct backend?

Reply:
yes for types Triton can do this